pczt/src/transparent.rs

```diff
@@ -0,0 +1,388 @@
+    ///   valid under the corresponding pubkey.
+    /// - These are required by the Spend Finalizer to assemble `script_sig`.
+    pub(crate) partial_signatures: BTreeMap<[u8; 33], Vec<u8>>,
+
+    /// The sighash type to be used for this input.
+    ///
+    /// - Signers must use this sighash type to produce their signatures. Signers that
+    ///   cannot produce signatures for this sighash type must not provide a signature.
+    /// - Spend Finalizers must fail to finalize inputs which have signatures that do not
+    ///   match this sighash type.
+    pub(crate) sighash_type: u8,
+
+    /// A map from a pubkey to the BIP 32 derivation path at which its corresponding
+    /// spending key can be found.
+    ///
+    /// - The pubkeys should appear in `script_pubkey` or `redeem_script`.
+    /// - Each entry is set by an Updater.
+    /// - Individual entries may be required by a Signer.
+    pub(crate) bip32_derivation: BTreeMap<[u8; 33], Zip32Derivation>,
+
+    /// Mappings of the form `key = RIPEMD160(value)`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) ripemd160_preimages: BTreeMap<[u8; 20], Vec<u8>>,
+
+    /// Mappings of the form `key = SHA256(value)`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) sha256_preimages: BTreeMap<[u8; 32], Vec<u8>>,
+
+    /// Mappings of the form `key = RIPEMD160(SHA256(value))`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) hash160_preimages: BTreeMap<[u8; 20], Vec<u8>>,
+
+    /// Mappings of the form `key = SHA256(SHA256(value))`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) hash256_preimages: BTreeMap<[u8; 32], Vec<u8>>,
+
+    /// Proprietary fields related to the note being spent.
+    pub(crate) proprietary: BTreeMap<String, Vec<u8>>,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct Output {
+    //
+    // Transparent effecting data.
+    //
+    // These are required fields that are part of the final transaction, and are filled in
+    // by the Constructor when adding an output.
+    //
+    pub(crate) value: u64,
+    pub(crate) script_pubkey: Vec<u8>,
+
+    /// The script required to spend this output, if it is P2SH.
+    ///
+    /// Set to `None` if this is a P2PKH output.
+    pub(crate) redeem_script: Option<Vec<u8>>,
+
+    /// A map from a pubkey to the BIP 32 derivation path at which its corresponding
+    /// spending key can be found.
+    ///
+    /// - The pubkeys should appear in `script_pubkey` or `redeem_script`.
+    /// - Each entry is set by an Updater.
+    /// - Individual entries may be required by a Signer.
+    pub(crate) bip32_derivation: BTreeMap<[u8; 33], Zip32Derivation>,
```

Comment:
I feel like I'm missing something here, why is this part of the `Output`?

Reply:
I would also like to know this!

Reply:
I think it's essentially the same reason as for inputs, i.e. there is no other way to identify the derivation paths for keys in `script_pubkey` or `redeem_script`.